### ISSUE TITLE
Crash in production app

### ISSUE THREAD

**Issue by infapednekar** [realm/realm-js#3496]
Steps for others to Reproduce
Not sure how to reproduce. This is happening in production app and is detected in Crashlytics logs.

From Crashlytics:
`
realm::Array::get(char const*, unsigned long) + 4240612
realm::StringData realm::ConstObj::_get<realm::StringData>(realm::ColKey::Idx) const + 304
object_accessor.hpp - Line 160
OpaqueJSValue const* realm::Object::get_property_value_impl<OpaqueJSValue const*, realm::js::NativeAccessor<realm::jsc::Types> >(realm::js::NativeAccessor<realm::jsc::Types>&, realm::Property const&) const + 160
jsc_return_value.hpp - Line 36
realm::js::RealmObjectClass<realm::jsc::Types>::get_property(OpaqueJSContext const_, OpaqueJSValue_, realm::js::String<realm::jsc::Types> const&, realm::js::ReturnValue<realm::jsc::Types>&) + 36
jsc_string.hpp - Line 41
OpaqueJSValue const* realm::js::wrap<&(realm::js::RealmObjectClass<realm::jsc::Types>::get_property(OpaqueJSContext const_, OpaqueJSValue_, realm::js::String<realm::jsc::Types> const&, realm::js::ReturnValue<realm::jsc::Types>&))>(OpaqueJSContext const_, OpaqueJSValue_, OpaqueJSString_, OpaqueJSValue const_*) + 41
`

IMPORTANT: Please send this log and info about Realm SDK version and other relevant reproduction info to help@realm.io.
Version of Realm and Tooling
Realm version:	10.1.0

/Applications/Xcode.app/Contents/Developer
Xcode 11.6

/usr/local/bin/pod
1.9.3

Complete Stacktrace:
[complete_stacktrace.txt](https://github.com/realm/realm-cocoa/files/5777080/stacktrace.txt)

**Comment by DominicFrei** [comment]
Hi @infapednekar, I guess you are using react-native? I've moved your issue to the corresponding repo (realm-js).

**Comment by kneth** [comment]
@infapednekar Thank you for reporting. Unfortunately it is very little information to find a bug. Is it possible to obtain the Realm file from the crash? Just so we can see if it is a data corruption case.

**Comment by infapednekar** [comment]
Hi @kneth . Thanks for reply. At the moment we are unable to reproduce it this crash in our local environment. We are only able to find this crash in crashlytics. I will share realm file once I am able to replicate the crash scenario on my device.

**Comment by infapednekar** [comment]
@kneth. Here are the realm log files for the device in which the crash occurred
[RealmLogs.zip](https://github.com/realm/realm-js/files/5832864/RealmLogs.zip)

**Comment by infapednekar** [comment]
> @kneth. Here are the realm log files for the device in which the crash occurred
> [RealmLogs.zip](https://github.com/realm/realm-js/files/5832864/RealmLogs.zip)

Hi @kneth. Did you had a chance to look into the logs.

**Comment by kneth** [comment]
@infapednekar `default.realm` can be opened locally. Unfortunately the stacktrace doesn't tell us which property, it is trying to get. We will examine `default.realm` a bit closer.

**Comment by kneth** [comment]
@infapednekar The Realm file isn't corrupted and we can't say what caused the crash. Do you have any idea of the context? I mean, can you give us a code snippet in where the crashed occurred?

**Comment by infapednekar** [comment]
@kneth One of the dev was able to reproduce the error on the ios device. Following is the scenario he provided - Seems realm DB is trying to insert an entry and not able to do so because the iPhone is locked with Passcode and hence db is encrypted. This was causing a crash.
 
```
Crashed: com.facebook.react.JavaScript
0 TSC 0x105043300 realm::Array::get(char const*, unsigned long) + 4436176
1 TSC 0x10514ce58 realm::StringData realm::ConstObj::_get<realm::StringData>(realm::ColKey::Idx) const + 304
2 TSC 0x104b89374 OpaqueJSValue const* realm::Object::get_property_value_impl<OpaqueJSValue const*, realm::js::NativeAccessor<realm::jsc::Types> >(realm::js::NativeAccessor<realm::jsc::Types>&, realm::Property const&) const + 160 (object_accessor.hpp:160)
3 TSC 0x104b94100 realm::js::RealmObjectClass<realm::jsc::Types>::get_property(OpaqueJSContext const*, OpaqueJSValue*, realm::js::String<realm::jsc::Types> const&, realm::js::ReturnValue<realm::jsc::Types>&) + 36 (jsc_return_value.hpp:36)
4 TSC 0x104b937f4 OpaqueJSValue const* realm::js::wrap<&(realm::js::RealmObjectClass<realm::jsc::Types>::get_property(OpaqueJSContext const*, OpaqueJSValue*, realm::js::String<realm::jsc::Types> const&, realm::js::ReturnValue<realm::jsc::Types>&))>(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSString*, OpaqueJSValue const**) + 41 (jsc_string.hpp:41)
5 JavaScriptCore 0x193fe71d8 JSC::JSCallbackObject<JSC::JSNonFinalObject>::call(JSC::JSGlobalObject*, JSC::CallFrame*) + 420 6 JavaScriptCore 0x193f6ff44 vmEntryToNative + 276
```

**Comment by kneth** [comment]
@infapednekar We have observed similar issues, for examplehttps://github.com/realm/realm-js/issues/3162#issuecomment-683203886

**Comment by kneth** [comment]
We have released v11.0.0 which has a complete rewrite of the integration with JSC. I am closing the issue, and ask you to upgrade. If the issue is still observed, please create a new issue.
